spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -555,38 +555,154 @@ of these methods. If it knows it will never be able to compensate or complete th
 
-Similarly if it cannot perform a completion activity immediately.
+==== Participant marker annotations method signatures
 
-If the `@Compensate` or `@Complete` annotation is present on multiple methods
-then an arbitrary one is chosen. If the annotation is not accompanied by
-a JAX-RS `@PUT` annotation the error SHOULD be reported using a JAX-RS
+The participant marker annotations are annotations that allow users to mark a method 
+for the execution by the LRA implemenatation according to the <<participant-state-model,
+the participant state model>>. These annotations are: 
+
+* `@Compensate` -- a method to be executed when the LRA is cancelled
+* `@Complete` -- a method to be executed when the LRA is completed
+* `@Status` -- a method that allow user to state status of the participant with regards 
+to a paricular LRA
+* `@Forget` -- a method to be executed when the LRA allows participant to 
+clear all associated information
+
+This specification is generally expected to be associated with the JAX-RS runtime. However, 
+we acknowledge that the mapping to JAX-RS resources may not always be feasible which is 
+why we distinguish two types of participant method definitions -- associated with the 
+JAX-RS resource method or not bound to JAX-RS. 
+
+[[jaxrs-participant-methods]]
+===== JAX-RS participant methods
+
+The following table presents expectations that are placed on individual participant 
+annotations when associated with JAX-RS resource methods:
+
+[[jaxrs-response-table]]
+|===
+| Annotation | HTTP method | Expected status codes | Response
+
+| `@Compensate`
+| PUT
+| 200, 202, 404, 412
+| <<source-ParticipantStatus,ParticipantStatus>>
+
+| `@Complete`
+| PUT
+| 200, 202, 404, 412
+| <<source-ParticipantStatus,ParticipantStatus>>
+
+| `@Status`
+| GET
+| 200, 202, 404, 412
+| <<source-ParticipantStatus,ParticipantStatus>>
+
+| `@Forget`
+| DELETE
+| 200, 404, 412
+| no expectations
+
+|=== 
+
+If the participant annotation is not accompanied by an associated JAX-RS HTTP method 
+annotation according to the <<jaxrs-response-table,table above>>, the error SHOULD be reported using a JAX-RS 
 exception mapper that maps to a `412 Precondition Failed` HTTP status code.
 
+[[non-jaxrs-participant-methods]]
+===== Non-JAX-RS participant methods
+
+When the participant annotations are applied to the non-JAX-RS resource methods they 
+MUST adhere to these predefined signatures:
+
+* *Return type*: 
+** `void`: successfull execution is mapped to `Compensated` or `Completed` participant statuses, 
+ error execution is handled by exception mapping
+*** applicable only for `@Compensate` and `@Complete` participant methods
+** <<source-ParticipantStatus,ParticipantStatus>>
+** `javax.ws.rs.core.Response`: handled similarly as for 
+<<jaxrs-participant-methods, JAX-RS participant methods>>
+* *Arguments*: up to 2 arguments of types:
+** `java.net.URI`: representing LRA id
+** `java.lang.String`: representing user data
```

Comment:
This is then something for 1.x already, as we haven't defined yet how you can supply these values? https://github.com/eclipse/microprofile-lra/issues/15

Reply:
thank you, I must have missed that we've moved it to 1.x. I will replace String userData with URI parentLraId and will require right order of arguments